app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,39 @@
+class MoviesController < ApplicationController
+
+  def index
+    movies = Movie.all
+    render json: movies.as_json(only: [:id, :title, :release_date]), status: :ok
+  end
+
+  def show
+    movie_id = params[:id]
+    movie = Movie.find_by(id: movie_id)
+
+    if movie
+      avail = movie.calculate_available_inventory
+      result = movie.save_available_inventory(avail)
```

Comment:
You end up assigning the result of `movie.save_available_inventory(avail)` into the variable `result`, but you don't use `result` anywhere... you probably don't need to assign this variable to anything